### DISCUSSION TITLE
e2e test 如何测试 service input

### DISCUSSION THREAD

**Discussion by snakorse** [alibaba/ilogtail#537] [Help] [answered]
比如 input_http_server 这种input，如何 构建http请求？

ilogtail-e2e.yaml中的trigger 好像不能发带body的post请求

**Comment by snakorse** [comment]
尝试修改了下test engine的代码，以使trigger 支持post https://github.com/alibaba/ilogtail/pull/521/commits/37f598f4bceda5ec244ac918869d4bac633257ab

**Reply by henryzhx8** [reply]
嗯，之前确实不支持body，这样的修改是可以的，是不是还没有正式PR？

**Reply by snakorse** [reply]
在这个pr里 https://github.com/alibaba/ilogtail/pull/521

**Comment by snakorse** [comment, marked as answer]
在进一步调试的过程中，发现如下问题:
1. 当定义了需要暴露port的input插件，则无法通过trigger访问到
    方案：ilogtail-e2e.yaml中 ilogtail节点增加配置：ports，允许额外暴露port
3. 测试flusher时，ilogtail-e2e.yaml中定义outputs节点后，实际运行会报未定义flusher配置
    方案：存在bug，ilogtail-e2e.yaml中指定的是 outputs节点，ilogtail pipeline配置最后需要的是 flushers 配置节点，缺少正确的转换

已在pr #521 中修复，commmit: https://github.com/alibaba/ilogtail/pull/521/commits/c00e41536f2b92a1f274e503b155d844fe3eeac2
